plugins/catalog-backend-module-bitbucket-server/src/providers/BitbucketServerEntityProvider.ts

```diff
@@ -407,19 +419,24 @@ export class BitbucketServerEntityProvider implements EntityProvider {
    * if not, it discovers any entity that was added and removed in the list of entities
    * @param event - A Bitbucket Server webhook event for repo:refs_change
    */
-  async onRepoPush(event: Events.PushEvent): Promise<void> {
+  async onRepoPush(
+    event: BitbucketServerEvents.RefsChangedEvent,
+  ): Promise<void> {
     if (!this.canHandleEvents()) {
+      this.logger.error(
+        'Bitbucket Server catalog entity provider is not set up to handle events. Missing tokenManager or catalogApi.',
+      );
       return;
     }
 
     if (!this.connection) {
       throw new Error('Not initialized');
     }
 
-    if (!this.isHeadPush(event)) return;
+    if (this.DEFAULT_BRANCH !== undefined && !this.isDefaultBranchPush(event))
+      return;
```

Comment:
```suggestion
    if (this.DEFAULT_BRANCH !== undefined && !this.isDefaultBranchPush(event)) {
      return;
    }
```

to use the same code style as at the if-blocks above